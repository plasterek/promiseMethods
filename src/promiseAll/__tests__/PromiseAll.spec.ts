import { promiseAll } from "../PromiseAll.service";

describe("PromiseAll function", () => {
  const resolvedPromise: () => Promise<unknown> = () => Promise.resolve("resolved");
  const rejectedPromise: () => Promise<unknown> = () => Promise.reject("rejected");
  describe("When every promise in given array is resolved", () => {
    it("It returns Promise with array of given promise results", () => {
      //given
      const promiseArray: (() => Promise<unknown>)[] = [resolvedPromise, resolvedPromise];
      //when
      const results: Promise<unknown> = promiseAll(promiseArray);
      //then
      expect.assertions(1);
      expect(results).resolves.toMatchSnapshot();
    });
  });
  describe("When any promise rejects in given array", () => {
    it("It returns Promise with rejected value", () => {
      //given
      const promiseArray: (() => Promise<unknown>)[] = [resolvedPromise, rejectedPromise];
      //when
      const results: Promise<unknown> = promiseAll(promiseArray);
      //then
      expect.assertions(1);
      expect(results).rejects.toMatchSnapshot();
    });
  });
  describe("When given array is empty", () => {
    it("It returns Promise with empty array", () => {
      //given
      const promiseArray: (() => Promise<unknown>)[] = [];
      //when
      const results: Promise<unknown> = promiseAll(promiseArray);
      //then
      expect.assertions(1);
      expect(results).resolves.toMatchSnapshot();
    });
  });
});
