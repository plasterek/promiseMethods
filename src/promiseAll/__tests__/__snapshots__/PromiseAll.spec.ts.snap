// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromiseAll function When any promise rejects in given array It returns Promise with rejected value 1`] = `"rejected"`;

exports[`PromiseAll function When every promise in given array is resolved It returns Promise with array of given promise results 1`] = `
[
  "resolved",
  "resolved",
]
`;

exports[`PromiseAll function When given array is empty It returns Promise with empty array 1`] = `[]`;
